packages/app/src/cli/services/dev/app-events/app-event-watcher.ts

```diff
@@ -0,0 +1,279 @@
+  const deletedExtensions = oldExtensions.filter((ext) => !newExtensionsHandles.includes(ext.handle))
+
+  const updatedExtensions = newExtensions.filter((ext) => {
+    const oldConfig = oldExtensions.find((oldExt) => oldExt.handle === ext.handle)?.configuration
+    const newConfig = ext.configuration
+    if (oldConfig === undefined) return false
+    return JSON.stringify(oldConfig) !== JSON.stringify(newConfig)
+  })
+  const createdEvents = createdExtensions.map((ext) => ({type: EventType.Created, extension: ext}))
+  const deletedEvents = deletedExtensions.map((ext) => ({type: EventType.Deleted, extension: ext}))
+  const updatedEvents = updatedExtensions.map((ext) => ({type: EventType.UpdatedSourceFile, extension: ext}))
+  return {
+    app: newApp,
+    extensionEvents: [...createdEvents, ...deletedEvents, ...updatedEvents],
+    startTime: event.startTime,
+    path: event.path,
+  }
+}
+
+/**
+ * When the app.toml is updated:
+ * Reload the app and return the new app and the updated extensions in the event.
+ * Compare the old and new extensions (defined in the app tomle) to find the created, deleted and updated extensions.
+ */
+async function AppConfigUpdatedHandler({event, app, options}: HandlerInput): Promise<AppEvent> {
+  const newApp = await reloadApp(app, options)
+  const oldExtensions = app.realExtensions.filter((ext) => ext.configurationPath === app.configuration.path)
+  const oldExtensionsHandles = oldExtensions.map((ext) => ext.handle)
+  const newExtensions = newApp.realExtensions.filter((ext) => ext.configurationPath === app.configuration.path)
+  const newExtensionsHandles = newExtensions.map((ext) => ext.handle)
+
+  const createdExtensions = newExtensions.filter((ext) => !oldExtensionsHandles.includes(ext.handle))
+  const deletedExtensions = oldExtensions.filter((ext) => !newExtensionsHandles.includes(ext.handle))
+  const updatedExtensions = newExtensions.filter((ext) => {
+    const oldConfig = oldExtensions.find((oldExt) => oldExt.handle === ext.handle)?.configuration
+    const newConfig = ext.configuration
+    if (oldConfig === undefined) return false
+    return JSON.stringify(oldConfig) !== JSON.stringify(newConfig)
+  })
+  const createEvents = createdExtensions.map((ext) => ({type: EventType.Created, extension: ext}))
+  const deleteEvents = deletedExtensions.map((ext) => ({type: EventType.Deleted, extension: ext}))
+  const updateEvents = updatedExtensions.map((ext) => ({type: EventType.Updated, extension: ext}))
+  return {
+    app: newApp,
+    extensionEvents: [...createEvents, ...deleteEvents, ...updateEvents],
+    startTime: event.startTime,
+    path: event.path,
+  }
+}
+
+/**
+ * When the app.toml is deleted:
+ * Throw an error to exit the process.
+ */
+async function AppConfigDeletedHandler(_input: HandlerInput): Promise<AppEvent> {
+  // The user deleted the active app.toml, why would they do that? :(
```

Comment:
If I rename the app config file -- that'd abort the CLI?

Reply:
Yeah, that will count as a deletion. 
We could support that use case, but is it worth it?

Reply:
I don't think its worth it.